### PR TITLE
Changes needed for rocAL MaskRCNN training

### PR DESCRIPTION
**Pull request by SundarRajan28**
- AMD OpenVX changes for adding ResizeMirrorNormalizePD node
- Adding an flag parameter to all meta nodes in rocAL for modifying metadata depending on the flag
- Metadata changes in rocAL
- Changing bbox datatype to double and adding the necessary rocAL and pybind changes
- rocAL Pybind changes for RMN augmentation and reader changes
- Adding a new example python pipeline which can be run to check the segmentation pipeline

### CODE REVIEW

File: rocAL/rocAL/include/api/rocal_api_info.h [hunk 1]
```diff
@@ -33,6 +33,18 @@ extern "C" int ROCAL_API_CALL rocalGetOutputWidth(RocalContext rocal_context);
 /// \return The height of the ROCAL's output image in pixels. It includes all images in the batch.
 extern "C" int ROCAL_API_CALL rocalGetOutputHeight(RocalContext rocal_context);
 
+///
+/// \param rocal_context
+/// \param buf
+/// \return The width of the ROCAL's output image in pixels
+extern "C" void ROCAL_API_CALL rocalGetOutputResizeWidth(RocalContext rocal_context, unsigned int* buf);
```
Comment (shobana-mcw) [inline rocAL/rocAL/include/api/rocal_api_info.h]: Why do we need this API?

File: rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize_mirror_normalize.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+void ResizeMirrorNormalizeNode::update_node()
+{
+    std::vector<uint32_t> src_roi_width, src_roi_height;
+    src_roi_width = _inputs[0]->info().get_roi_width_vec();
+    src_roi_height = _inputs[0]->info().get_roi_height_vec();
+
+    for(uint i = 0; i < _batch_size; i++)
+    {
+        int min_size = 800;
```
Comment (shobana-mcw) [inline rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize_mirror_normalize.cpp:96]: Why are we hardcoding these values?

File: rocAL/rocAL/source/pipeline/image.cpp [hunk 3]
```diff
@@ -132,7 +132,7 @@ ImageInfo::ImageInfo(
         _height(height_),
         _color_planes(planes),
         _batch_size(batches),
-        _data_size((static_cast<uint64_t>(width_ * height_ * _batch_size * planes))),
+        _data_size((static_cast<uint64_t>(width_ * height_ * _batch_size * planes / 8) * 8 + 8)),
```
Comment (shobana-mcw) [inline rocAL/rocAL/source/pipeline/image.cpp:147]: Why are we doing this?

**Review by shobana-mcw — COMMENTED** [review]
I wanted to discuss with you about handling the segmentation flag in meta part.

File: rocAL/rocAL/source/pipeline/image.cpp (hunk 3 above)
Comment (SundarRajan98) [reply rocAL/rocAL/source/pipeline/image.cpp:147]: @sampath1117 Can u explain why this patch is needed here?

File: rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize_mirror_normalize.cpp (hunk 2 above)
Comment (SundarRajan98) [reply rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize_mirror_normalize.cpp:96]: The min and max size is constant so it was hardcoded to this node

File: rocAL/rocAL/include/api/rocal_api_info.h (hunk 1 above)
Comment (SundarRajan98) [reply rocAL/rocAL/include/api/rocal_api_info.h]: Used to get the resize ROI width and height in python

**Pull request closed without merging**